src/Symfony/Component/Workflow/Workflow.php

```diff
@@ -59,6 +59,10 @@ public function getMarking($subject)
     {
         $marking = $this->markingStore->getMarking($subject);
 
+        if (!$marking instanceof Marking) {
+            throw new LogicException(sprintf('The value returned by the MarkingStore is not an instance of "%s" for workflow "%s".', Marking::class, $this->name));
```

Comment:
For being a contract check, i guess `LogicException` is just fine also :) (opposed to `UnexpectedValue`)